internal/flags/time_value.go

```diff
@@ -0,0 +1,46 @@
+package flags
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	TimeLayout     = time.RFC3339
+	TimeLayoutName = "RFC3339"
+)
+
+// TimeValue is a custom [pflag.Value] implementation for [time.Time]
+// that provides a better formatted error message using [time.RFC3339] layout.
+type TimeValue struct {
+	value *time.Time
+}
+
+func (t *TimeValue) String() string {
+	if t.value == nil || t.value.IsZero() {
+		return ""
+	}
+	return t.value.Format(TimeLayout)
+}
+
+func (t *TimeValue) Set(s string) error {
+	parsed, err := time.Parse(TimeLayout, s)
+	if err != nil {
+		return fmt.Errorf(
```

Comment:
nitpick: 
```suggestion
		return errors.New(
```

or prev implementation

Reply:
https://github.com/nobl9/sloctl/pull/396#discussion_r2627615601